### PR TITLE
Xeno Tweaks and Fixes

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Fixing up Xenos, one small PR at a time.

- Implements TG's icon cache for xeno weeds
- Fixes up some behaviors with weed/eggs
- Acid can now melt through the asteroid and minerals
- Xenos now play a gib sound when...gibbing
- Removed a duplicate icon 

:cl: Fox McCloud
bugfix: Fixes up some behaviors with xeno eggs
tweak: Xeno acid can now melt through floors and the asteroid
rscadd: Xeno's now play the gib sound when actually gibbed
/:cl:

### CODE REVIEW

File: code/game/objects/effects/aliens.dm [hunk 1]
```diff
@@ -229,25 +235,35 @@
...
 /obj/structure/alien/weeds/proc/updateWeedOverlays()
 
 	overlays.Cut()
+
+	if(!weedImageCache || !weedImageCache.len)
+		weedImageCache = list()
+		weedImageCache.len = 4
+		weedImageCache[WEED_NORTH_EDGING] = image('icons/mob/alien.dmi', "weeds_side_n", layer=2.11, pixel_y = -32)
+		weedImageCache[WEED_SOUTH_EDGING] = image('icons/mob/alien.dmi', "weeds_side_s", layer=2.11, pixel_y = 32)
+		weedImageCache[WEED_EAST_EDGING] = image('icons/mob/alien.dmi', "weeds_side_e", layer=2.11, pixel_x = -32)
```
Comment (tigercat2000) [inline code/game/objects/effects/aliens.dm]: c-could you please just like
add a single space to these bottom ones
so all the image commands line up perfectly

File: code/game/objects/effects/aliens.dm [hunk 2]
```diff
@@ -297,9 +314,8 @@
 	spawn(rand(MIN_GROWTH_TIME, MAX_GROWTH_TIME))
 		Grow()
 
-
-/obj/structure/alien/egg/attack_alien(mob/user)
-	if(isalien(user))
+/obj/structure/alien/egg/attack_hand(mob/living/user)
```
Comment (tigercat2000) [inline code/game/objects/effects/aliens.dm]: don't believe aliens can call attack_hand

Comment (Fox-McCloud) [inline code/game/objects/effects/aliens.dm]: They do, actually!

https://github.com/ParadiseSS13/Paradise/blob/3f72596e62eb7313a11bb02ffe78e481a3efddef/code/_onclick/other_mobs.dm#L64

Comment (tigercat2000) [inline code/game/objects/effects/aliens.dm]: How odd.

**Pull request merged**